graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionProtocolHandler.kt

```diff
@@ -144,14 +144,22 @@ class ApolloSubscriptionProtocolHandler(
     }
 
     private fun onInit(operationMessage: SubscriptionOperationMessage, session: WebSocketSession, graphQLContext: Any?): Flux<SubscriptionOperationMessage> {
-        val connectionParams = operationMessage.payload as? Map<String, String> ?: emptyMap()
+        val connectionParams = convertPayloadToMap(operationMessage.payload)
         val onConnect = subscriptionHooks.onConnect(connectionParams, session, graphQLContext)
         sessionState.saveOnConnectHook(session, onConnect)
         val acknowledgeMessageFlux = Flux.just(acknowledgeMessage)
         val keepAliveFlux = getKeepAliveFlux(session)
         return acknowledgeMessageFlux.concatWith(keepAliveFlux)
     }
 
+    @Suppress("UNCHECKED_CAST")
+    private fun convertPayloadToMap(payload: Any?): Map<String, String> =
+        if (payload is Map<*, *>) {
```

Comment:
whats the point of extra `is` check? e.g. why not just keep it simply `payload as? Map<String, String> ?: emptyMap()` as it used to be (and in that case do we really need one liner method?)?

Reply:
Yea, will delete. This was added so I could verify what the warning in intellij was complaining about.